sdk/src/logs/logger.cc

```diff
@@ -36,23 +51,76 @@ void Logger::log(const opentelemetry::logs::LogRecord &record) noexcept
 
   // TODO: Sampler logic (should include check for minSeverity)
 
-  /**
-   * Convert the LogRecord to the heap first before sending to processor.
-   * TODO: Change the API log(LogRecord) function to log(*LogRecord) so the following line
-   * converting record a heap variable can be removed
-   */
-  auto record_pointer =
-      std::unique_ptr<opentelemetry::logs::LogRecord>(new opentelemetry::logs::LogRecord(record));
+  auto recordable = processor->MakeRecordable();
```

Comment:
I do realize that we allow the SDK built with or without exceptions. However, if we build it with exceptions, is there a provision for the `processor->MakeRecordable()` to throw `std::bad_alloc`? What happens here, is in environment where exceptions are enabled - we'd be silently dropping telemetry. This may not be good in some cases, i.e. if it's a security audit log - we need to potentially bubble up the error condition to the caller, for the caller to decide what to do with that.

I do not think that this is blocking. But if you can log an issue about this, that'd work fine.

Reply:
Okay, I see that this should be handled by new in `MakeRecordable()` method potentially throwing. Please ignore my above comment.

Reply:
If recordable is null, maybe an error could be invoked using #408?

Reply:
If the processor returns a `nullptr`, could we note an error like:
 ```
if (recordable == nullptr) { 
     error "recordable creation failed";
     return;
}
```` 
where the error is handled by error diagnostics in #408? We may not know the reason for it if the processor/exporter handled it, but at least the user will know something went wrong

Reply:
This could be a workaround. Could you add an `TODO` in the `if` check to add such error diagnostic?

Reply:
Sure, added `TODO` and just opened up https://github.com/open-telemetry/opentelemetry-cpp/issues/441 for further discussion